### ISSUE TITLE
Table alias in a new line causing LT02 failure

### ISSUE THREAD

**Issue by gvrvpndt** [sqlfluff/sqlfluff#5639]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff throws `LT02` error for a seemingly valid sql:

```sql
SELECT
  Table1.id1,
  Table1.id2,
  Table2.col2
FROM some_very_loooooooooong_qualifier.LongNameTable1 AS Table1
INNER JOIN some_very_loooooooooooooooooooooooooooooong_qualifier.LongNameTable2
    AS Table2       # table aliasing is on the new line because above line it too long already
  ON Table1.id1 = Table2.id1
    AND Table1.id2 = Table2.id2
```

### Expected Behaviour

No lint error.

### Observed Behaviour

sqlfluff throws the following lint error:
```
L:   6 | P:  11 | LT02 | Expected line break and indent of 2 spaces before 'some_very_loooooooooooooooooooooooooooooong_qualifier'. [layout.indent]                                                                                                                                       
L:   8 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   9 | P:   1 | LT02 | Expected indent of 6 spaces. [layout.indent]
```

### How to reproduce

Run above sql script through sqlfluff linter.

### Dialect

bigquery

### Version

sqlfluff version 2.3.5
python version 3.9.13

### Configuration

[sqlfluff]
dialect = bigquery
templater = jinja

exclude_rules = aliasing.expression, capitalisation.identifiers, layout.cte_newline, references.keywords, aliasing.forbid, structure.using, structure.column_order, structure.else_null, layout.select_targets, convention.terminator, layout.end_of_file
output_line_length = 250
max_line_length = 250
sql_file_exts = .sql

[sqlfluff:indentation]
tab_space_size = 2
indented_ctes = True
allow_implicit_indents = True

[sqlfluff:rules]
allow_scalar = False

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
unquoted_identifiers_policy = column_aliases

[sqlfluff:rules:references.keywords]
unquoted_identifiers_policy = all
quoted_identifiers_policy = none

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:convention.terminator]
require_final_semicolon = True

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:convention.quoted_literals]
preferred_quoted_literal_style = single_quotes

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by gvrvpndt** [comment]
To add, I can make sqlfluff happy with either of the two changes described below, but we are not in position to update 100s of sql files. I am looking for a way to change the config itself so that we can continue to be able to alias on a new line.

1. Keep table alias on the same line:
```
SELECT
  ...
FROM some_very_loooooooooong_qualifier.LongNameTable1 AS Table1
INNER JOIN some_very_loooooooooooooooooooooooooooooong_qualifier.LongNameTable2 AS Table2
  ON ...
```
2. Move joined table to a new line:
```
SELECT
  ...
FROM some_very_loooooooooong_qualifier.LongNameTable1 AS Table1
INNER JOIN
  some_very_loooooooooooooooooooooooooooooong_qualifier.LongNameTable2
    AS Table2
  ON ...
```

**Comment by peterbud** [comment]
Hi @gvrvpndt - thanks for opening this issue.

can you try the below configration setting:
```ini
[sqlfluff:layout:type:join_clause]
keyword_line_position = leading
```

I have tested and LT02 is not being triggered with the original statement.

**Comment by gvrvpndt** [comment]
@peterbud Unfortunately, I won't be able to test this any time soon. I think your testing makes sense and this can be closed if that worked.
